pkg/operator/ceph/cluster/cluster.go

```diff
@@ -598,31 +598,41 @@ func (c *cluster) reportTelemetry() {
 
 func (c *cluster) configureMsgr2() error {
 	encryptionSetting := "secure"
-	defaultRBDMapOptions := "ms_mode=secure"
+	rbdMapOptions := "rbd_default_map_options"
 	encryptionGlobalConfigSettings := map[string]string{
-		"ms_cluster_mode":         encryptionSetting,
-		"ms_service_mode":         encryptionSetting,
-		"ms_client_mode":          encryptionSetting,
-		"rbd_default_map_options": defaultRBDMapOptions,
+		"ms_cluster_mode": encryptionSetting,
+		"ms_service_mode": encryptionSetting,
+		"ms_client_mode":  encryptionSetting,
+		rbdMapOptions:     "ms_mode=secure",
 	}
 	monStore := config.GetMonStore(c.context, c.ClusterInfo)
 
-	// If the user has disabled encryption, remove the encryption settings
-	if c.Spec.Network.Connections == nil || c.Spec.Network.Connections.Encryption == nil || !c.Spec.Network.Connections.Encryption.Enabled {
+	encryptionEnabled := c.Spec.Network.Connections != nil &&
+		c.Spec.Network.Connections.Encryption != nil &&
+		c.Spec.Network.Connections.Encryption.Enabled
+
+	if encryptionEnabled {
+		logger.Infof("setting msgr2 encryption mode to %q", encryptionSetting)
+		if err := monStore.SetAll("global", encryptionGlobalConfigSettings); err != nil {
+			return err
+		}
+	} else {
 		encryptionConfig := []config.Option{}
 		for k := range encryptionGlobalConfigSettings {
 			encryptionConfig = append(encryptionConfig, config.Option{Who: "global", Option: k})
 		}
 		if err := monStore.DeleteAll(encryptionConfig...); err != nil {
```

Comment:
Before these settings are deleted, should the `rbdMapOptions` be removed from the map? Anyway it will immediately be set to `prefer-crc` on line 630 below, but seems like no need to delete first.

Reply:
> Anyway it will immediately be set to `prefer-crc` on line 630 below

Not unconditionally -- only if `requireMsgr2` is set.

Reply:
Yes as Ilya mentioned it will be set only if v2 is enabled but not encryption